### ISSUE TITLE
[Question] Is TNS compatible with Xcode 8?

### ISSUE THREAD

**Issue by NordlingArt** [NativeScript/NativeScript#2780]
I just recently installed macOS Sierra, but haven't updated Xcode to 8 yet. Still on 7.3 because I am unsure about the compatibility between NativeScript and Xcode 8. So the question is, am I good to proceed updating Xcode?

**Comment by enchev** [comment]
Hey @NordlingArt,

Yep, it is however you need to update everything to the latest versions (tns-core-modules, ios runtime, CLI, etc.). Apple changed a lot of methods to properties + introduced new code signing policy.

**Comment by NordlingArt** [comment]
@enchev - Thanks for letting me know. I have now upgraded to Xcode 8. I stumbled upon an issue that prevented my app to launch properly. I found the issue:

`UIApplication.sharedApplication()` is no longer a function. Apparently it should now be called with: `UIApplication.sharedApplication`

I still think this weird though since my physical iPhone 6 (iOS 10) runs the app with no complication.

**Comment by NickIliev** [comment]
The only difference when building iOS app with XCode 8 is that you will have to set your Team ID when building on real device. No changes needed if you are running your app in simulator.

If you are using XCode 8 and you want to buld on real device you have two options:
Option 1.)
Open **build.xcconfig**
```
// To build for device with XCode 8 you need to specify your development team. More info: https://developer.apple.com/library/prerelease/content/releasenotes/DeveloperTools/RN-Xcode/Introduction.html
// DEVELOPMENT_TEAM = YOUR_TEAM_ID;
```
As specified in the file comment you wil have to uncomment this line and set your team id (you can see your Team IDs in your Keychain Access app)

Option 2.)
This option is available with the @next modules (will ship with 2.4.0)
Instead of manually setting your build.xcconfig you can pass the team id with the CLI command.

```
tns build ios --teamId <your-team-id>
```
again notice that this option will run with tns-core-modules@next or after release 2.4.0


Resources on Apple code signing :
[https://developer.apple.com/library/prerelease/content/releasenotes/DeveloperTools/https://developer.apple.com/library/prerelease/content/releasenotes/DeveloperTools/RN-Xcode/Introduction.html-Xcode/Introduction.html](url)
 [https://developer.apple.com/videos/play/wwdc2016/401/](https://developer.apple.com/videos/play/wwdc2016/401/)
